pkg/operator/k8sutil/deployment.go

```diff
@@ -47,12 +48,18 @@ func GetDeploymentSpecImage(clientset kubernetes.Interface, d apps.Deployment, c
 
 // UpdateDeploymentAndWait updates a deployment and waits until it is running to return. It will
 // error if the deployment does not exist to be updated or if it takes too long.
-func UpdateDeploymentAndWait(context *clusterd.Context, deployment *apps.Deployment, namespace string) (*v1.Deployment, error) {
+func UpdateDeploymentAndWait(context *clusterd.Context, deployment *apps.Deployment, namespace string, verifyCallback func(action string) error) (*v1.Deployment, error) {
 	original, err := context.Clientset.AppsV1().Deployments(namespace).Get(deployment.Name, metav1.GetOptions{})
 	if err != nil {
 		return nil, fmt.Errorf("failed to get deployment %s. %+v", deployment.Name, err)
 	}
 
+	// Let's verify the deployment can be stopped
+	err = verifyCallback("stop")
+	if err != nil {
+		return nil, fmt.Errorf("failed to check if deployment %s can be updated: %+v", deployment.Name, err)
```

Comment:
I'm still struggling with this behavior of cancelling the orchestration if it's not ok to stop. The method is called `UpdateDeploymentAndWait()`. So it is expected that we will try and wait until it is safe to update the daemon and then continue. Why not just wait and retry for some timeout?

Reply:
Ok let's loop just like the following.